cvat-ui/src/components/annotation-page/standard-workspace/controls-side-bar/tools-control.tsx

```diff
@@ -1197,173 +1196,69 @@ export class ToolsControlComponent extends React.PureComponent<Props, State> {
-                                        };
-                                    }
-                                    return acc;
-                                }
-                                case 'checkbox':
-                                    return {
-                                        ...acc,
-                                        [attributeSpec.id as number]: `${value}`.toLowerCase() === 'true' ? 'true' : 'false',
-                                    };
-                                default:
-                                    return acc;
-                            }
-                        }, {} as Record<number, string>);
+                        return Object.fromEntries(attributes.map((a) => [a.spec_id, a.value]));
                     }
 
                     try {
                         this.setState({ mode: 'detection', fetching: true });
 
-                        // The function call endpoint doesn't support the cleanup and conv_mask_to_poly parameters.
-                        const { cleanup, conv_mask_to_poly: convMaskToPoly, ...restOfBody } = body;
+                        // The function call endpoint doesn't support the cleanup parameter.
+                        const { cleanup, ...restOfBody } = body;
 
                         const result = await core.lambda.call(jobInstance.taskId, model, {
                             ...restOfBody, frame, job: jobInstance.id,
-                        }) as DetectedShapes;
+                        }) as DetectorResults;
+
+                        const tagStates = result.tags.map((tag) => {
+                            const jobLabel = jobInstance.labels
+                                .find((jLabel) => jLabel.id === tag.label_id)!;
+
+                            return new core.classes.ObjectState({
+                                attributes: loadAttributes(tag.attributes),
+                                frame,
+                                label: jobLabel,
+                                objectType: ObjectType.TAG,
+                                source: core.enums.Source.AUTO,
+                            });
+                        });
 
-                        const states = result.map(
-                            (data): ObjectState | null => {
-                                const jobLabel = jobInstance.labels
-                                    .find((jLabel: Label): boolean => jLabel.name === data.label);
+                        const shapeStates = result.shapes.map((shape) => {
+                            const jobLabel = jobInstance.labels
+                                .find((jLabel) => jLabel.id === shape.label_id)!;
 
-                                if (!jobLabel) return null;
+                            return new core.classes.ObjectState({
+                                attributes: loadAttributes(shape.attributes),
+                                elements: shape.elements?.map?.((element) => {
```

Comment:
```suggestion
                                elements: shape.elements?.map((element) => {
```

Reply:
I don't think that's right. If `elements` is undefined, then `elements?.map` is also undefined, so calling it will throw an exception.

Reply:
Not really
If elements is undefined, the expression will return `undefined` immediately. It will not try to go further

```
test = {
    a: {
        b: {
            
        }
    }
}
{a: {…}}
test.a.b?.c?.d?.e?.f
    
>> undefined
```

Reply:
Oh, I see. Updated.